string who
string what
string where
---
bool ok

